src/commands/tileindex-validate/tileindex.validate.ts

```diff
@@ -357,6 +361,43 @@ export interface TiffLocation {
   bands: string[];
 }
 
+/**
+ * Calculate the number grid tiles touched by a TIFF with min being the lowest width/height, and max the largest.
+ * @param inputMin lowest extent of misaligned tile on one dimension (left or bottom)
+ * @param inputMax largest extent of misaligned tile on same dimension (right or top)
+ * @param gridOrigin grid origin (x or y corresponding to min/max dimension)
+ * @param tileSize height or width of the target grid tile (corresponding to min/max dimension)
+ */
+const calculateSteps = (inputMin: number, inputMax: number, gridOrigin: number, tileSize: number): number => {
+  const startStep = Math.floor((inputMin - gridOrigin) / tileSize);
+  const endStep = Math.floor((inputMax - gridOrigin - 1) / tileSize);
+
+  return endStep - startStep + 1;
+};
+
+/**
+ * Reproject the bounding box if the source and target projections are different.
+ * @param bbox input bounding box
+ * @param sourceProjection CRS of the input bounding box
+ * @param targetProjection target CRS
+ */
+function reprojectIfNeeded(
+  bbox: BBox,
+  sourceProjection: Projection,
+  targetProjection: Projection,
+): [number | undefined, number | undefined, number | undefined, number | undefined] {
+  {
+    if (targetProjection !== sourceProjection) {
+      const [ulX, ulY] = targetProjection.fromWgs84(sourceProjection.toWgs84([bbox[0], bbox[3]]));
+      const [lrX, lrY] = targetProjection.fromWgs84(sourceProjection.toWgs84([bbox[2], bbox[1]]));
+
+      return [ulX, lrY, lrX, ulY];
```

Comment:
is this `[minX, minY, maxX, maxY]` we generally do Math.min/Math.max here.

Reply:
Yes and no - I guess by doing this here without Math.min/Math.max we keep the order of the input intact, rather than taking it and possibly re-shuffling it by doing min/max.
It feels a tad safer than potentially re-ordering the data, especially as the other branch of the if statement does returns unchanged input bbox.

Reply:
after reprojecting the top left may no longer be the minX, maxY, if we are being consistent that a bbox is minX, minY, maxX, maxY we need to confirm it.

Reply:
done